src/main/java/frc/robot/subsystems/elevator/Elevator.java

```diff
@@ -0,0 +1,94 @@
+package frc.robot.subsystems.elevator;
+
+import frc.robot.hardware.digitalinput.DigitalInputInputsAutoLogged;
+import frc.robot.hardware.digitalinput.IDigitalInput;
+import frc.robot.hardware.motor.ControllableMotor;
+import frc.robot.hardware.request.IRequest;
+import frc.robot.hardware.request.cansparkmax.SparkMaxDoubleRequest;
+import frc.robot.subsystems.elevator.factories.RealElevatorConstants;
+import frc.utils.GBSubsystem;
+import org.littletonrobotics.junction.Logger;
+
+public class Elevator extends GBSubsystem {
+
+	private final DigitalInputInputsAutoLogged digitalInputsInputs;
+	private final ElevatorCommandBuilder commandBuilder;
+	private final IRequest positionRequest;
+	private double targetPosition;
+
+	private final ElevatorStuff elevatorStuff;
+	private final ControllableMotor mainMotor;
+	private final IDigitalInput limitSwitch;
+	private final String digitalInputsLogPath;
+
+	public Elevator(ElevatorStuff elevatorStuff) {
+		super(elevatorStuff.logPath());
+
+		this.mainMotor = elevatorStuff.mainMotor();
+		this.limitSwitch = elevatorStuff.digitalInput();
+		this.digitalInputsLogPath = elevatorStuff.digitalInputsLogPath();
+
+		this.digitalInputsInputs = new DigitalInputInputsAutoLogged();
+		this.elevatorStuff = elevatorStuff;
+		this.commandBuilder = new ElevatorCommandBuilder(this);
+
+		this.positionRequest = new SparkMaxDoubleRequest(targetPosition, SparkMaxDoubleRequest.SparkDoubleRequestType.CURRENT, 0);
```

Comment:
you don't initialize targetPosition, use current instead of position, have a magic number and this entire line should be in real elevator constants 🤮🤮